pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractScalarTransformFunction.java

```diff
@@ -105,210 +105,298 @@ public TransformResultMetadata getResultMetadata() {
 
   @Override
   public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
-    final String[] stringValuesSV = _jsonFieldTransformFunction.transformToStringValuesSV(projectionBlock);
-    final int[] results = new int[projectionBlock.getNumDocs()];
-    for (int i = 0; i < results.length; i++) {
-      Object read = JSON_PARSER_CONTEXT.parse(stringValuesSV[i]).read(_jsonPath);
-      if (read == null) {
+    if (_intValuesSV == null) {
+      _intValuesSV = new int[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    }
+
+    String[] jsonStrings = _jsonFieldTransformFunction.transformToStringValuesSV(projectionBlock);
+    int numDocs = projectionBlock.getNumDocs();
+    for (int i = 0; i < numDocs; i++) {
+      Object result = null;
+      try {
+        result = JSON_PARSER_CONTEXT.parse(jsonStrings[i]).read(_jsonPath);
+      } catch (Exception ignored) {
```

Comment:
I think we should not be catching exceptions here mainly because JSON_PARSER_CONTEXT has been declared to suppress exceptions already:

```
  private static final ParseContext JSON_PARSER_CONTEXT =
      JsonPath.using(Configuration.defaultConfiguration().addOptions(Option.SUPPRESS_EXCEPTIONS));
```
If for some reason, it does throw exception even with SUPPRESS_EXCEPTIONS option turned on, then we would like to know about that (as that would be a bug). Is there a specific case that you encountered that made caching exception here necessary?

Reply:
The parser will throw exception when input json string is `"null"` (this is a valid json, and I would count this as a bug in `JsonPath` library) or invalid json (e.g. `"{1"`).
If default value is provided, we should return the default value when encountering exception so that user can still get the result for the valid values. When default value is not there, we will throw the exception.